### PR TITLE
More support for netcoreapp3.0

### PR DESCRIPTION
**Pull request by slang25**
See #377

### CODE REVIEW

File: src/Giraffe/Giraffe.fsproj [hunk 1]
```diff
@@ -41,23 +41,29 @@
...
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.0' ">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' != 'netcoreapp3.0' ">
+    <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.*" />
+    <PackageReference Include="Microsoft.AspNetCore.Diagnostics" Version="2.2.*" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication" Version="2.2.*" />
+    <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="3.0.0" />
```
Comment (slang25) [inline src/Giraffe/Giraffe.fsproj]: Should this be version `2.2.*`, as I assume `3.0.0` was to fix compatibility with `netcoreapp3.0` which would now be covered by the framework reference.

**Comment by slang25** [comment]
This is straddling major aspnetcore versions, that's my only real concern here. MVC just went to `netcoreapp3.0` only.

Another thing to note, here I've lowered the major version of `Microsoft.AspNetCore.Authorization` compared to the Giraffe 4.0 release.

**Comment by NinoFloris** [comment]
I recommend looking at 
https://twitter.com/khellang/status/1178403463778258944?s=19
To work well with package references (2.x) and framework references (3+) until 2.x goes out of support.

**Comment by slang25** [comment]
Thanks @NinoFloris, this PR looks consistent with that approach and is ready for review.

File: tests/Giraffe.Tests/Giraffe.Tests.fsproj [hunk 2]
```diff
@@ -45,11 +54,11 @@
   <ItemGroup Condition="'$(TargetFramework)' != 'netcoreapp3.0'">
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.2.0" />
   </ItemGroup>
-  
+
```
Comment (dustinmoris) [inline tests/Giraffe.Tests/Giraffe.Tests.fsproj]: Both these `ItemGroup`s are repleated here.

**Comment by NinoFloris** [comment]
Thanks @slang25!

**Pull request merged**